add truecolor support for tmux (from upstream master)

therealpxc opened the pull request:
This won't be necessary with tmux-2.2, but that release may be quite a while yet. The patch is taken from here: https://github.com/tmux/tmux/issues/112

Commit:
add truecolor support for tmux (from upstream master)

globin commented on pkgs/tools/misc/tmux/default.nix:
```diff
@@ -13,6 +13,12 @@ stdenv.mkDerivation rec {
 
+#  prePatch = ''
+#    sed -e '/256colors.pl/d;d;d' -i 112.diff
+#  '';
```
Is this intentional?

globin commented on pkgs/tools/misc/tmux/default.nix:
```diff
@@ -13,6 +13,12 @@ stdenv.mkDerivation rec {
+#    sed -e '/256colors.pl/d;d;d' -i 112.diff
+#  '';
+
+  patches = copyPathToStore ./tmux-truecolor-gh112.patch;
```
The patch does not contain anything

Commit:
fix an oops (empty patch file)

Commit:
clean up tmux/default.nix comments

therealpxc commented:
I am a dummy! I've replaced the empty patch file with the right one and cleaned up the unnecessary commented-out bits in tmux/default.nix.

Thanks for looking this over, and sorry I made such obvious boo-boos.

therealpxc commented:
Is this now acceptable?

jgillich commented:
Why exactly does this belong in nixpkgs? It's adding a feature to a package, not something we usually do. You can always override the package.

therealpxc commented:
It may not really belong in nixpkgs. It was a sort of first attempt at contributing. If we want to hew closer to upstream, that's fine.

assaf758 commented:
While trying 24bit-color support, I noticed there is some issue (when comparing to non-tmux terminal) - see http://ibin.co/2YAmCLcoBNRC.
Happens with two tests (see below) and two terminals (termite and gnome-terminal):
https://raw.githubusercontent.com/robertknight/konsole/master/tests/color-spaces.pl
https://github.com/gnachman/iTerm2/blob/master/tests/24-bit-color.sh
